src/azure-cli/azure/cli/command_modules/acs/commands.py

```diff
@@ -85,6 +86,7 @@ def load_command_table(self, _):
                          'Are you sure you want to perform this operation?')
         g.custom_command('upgrade-connector', 'k8s_upgrade_connector', is_preview=True)
         g.custom_command('use-dev-spaces', 'aks_use_dev_spaces')
+        g.custom_command('update', 'aks_update', supports_no_wait=True)
```

Comment:
this should be a generic updater, check out how [VM does](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/vm/commands.py#L237)

Reply:
I think it's a little different here, not all the options we updated are from AKS models. Should we still swith to generic updater?

Reply:
Yes, I understand the difference.
The guideline calls out you should always start with generic updater, which is one liner simple change and gives you lots of other things for free such as `tagging`, and then you can expose strong arguments like the "--attach/detach acr".
For context missing `update` command from AKS is a long over-due, and since you are at it, I suggest very much you just get it done and resolve https://github.com/Azure/azure-cli/issues/1149.

Reply:
That's actually one of the issues. When the cluster enables multi agent pools, the create_or_update call would also fail because agent pools updates have their own subresource APIs.

Could we keep it as custom command until we figured out how to fix #1149?